src/azure-cli-core/azure/cli/core/cloud.py

```diff
@@ -27,87 +27,107 @@ def __str__(self):
 class CannotUnregisterCloudException(Exception):
     pass
 
-class CloudEndpoint(object): # pylint: disable=too-few-public-methods
-    MANAGEMENT = 'management'
-    RESOURCE_MANAGER = 'resource_manager'
-    SQL_MANAGEMENT = 'sql_management'
-    GALLERY = 'gallery'
-    ACTIVE_DIRECTORY = 'active_directory'
-    ACTIVE_DIRECTORY_RESOURCE_ID = 'active_directory_resource_id'
-    ACTIVE_DIRECTORY_GRAPH_RESOURCE_ID = 'active_directory_graph_resource_id'
-
-class CloudSuffix(object): # pylint: disable=too-few-public-methods
-    SQL_SERVER_HOSTNAME = 'sql_server_hostname'
-    STORAGE_ENDPOINT = 'storage_endpoint'
-    KEYVAULT_DNS = 'keyvault_dns'
-    AZURE_DATALAKE_STORE_FILE_SYSTEM_ENDPOINT = 'azure_datalake_store_file_system_endpoint' #pylint: disable=line-too-long
-    AZURE_DATALAKE_ANALYTICS_CATALOG_AND_JOB_ENDPOINT = 'azure_datalake_analytics_catalog_and_job_endpoint' #pylint: disable=line-too-long
+class CloudEndpoints(object):
+    def __init__(self,
+                 management=None,
```

Comment:
I don't think it makes sense for them to default to public Azure...
In that scenario you would pass in all the endpoints you need. If some of them happen to be the same as public Azure then so be it.

Also, when I register a cloud, I would not expect for the values I didn't specify to be set to point to public Azure.